frame/support/src/traits.rs

```diff
@@ -2035,6 +2065,17 @@ pub trait Hooks<BlockNumber> {
 	/// The block is being finalized. Implement to have something happen.
 	fn on_finalize(_n: BlockNumber) {}
 
+	/// This will be run when the block is being finalized (before on finalize).
+	/// Implement to have something happen using the remaining weight.
+	/// Returns the weight used, passes it to the next on idle hook if exists.
```

Comment:
This is not quite accurate, right?
It's "will subtract the returned weight from the remaining weight and will pass the reduced weight into the next on_idle call".

Reply:
yes, returns the weight used, subtracts if from the current weight and passes that, will fix.